lib/topological_inventory/sync/inventory_upload_worker.rb

```diff
@@ -14,11 +17,154 @@ def persist_ref
         "topological-inventory-upload-processor"
       end
 
+      def worker_name
+        "Topological Inventory Inventory Upload Worker"
+      end
+
       def perform(message)
         payload = JSON.parse(message.payload)
         return unless payload["service"] == "topological-inventory"
 
-        logger.info("#{payload}")
+        openurl(payload["url"]) do |io|
+          untargz(io) do |file|
+            process_data(payload, file.read)
+          end
+        end
+      end
+
+      private
+
+      def process_data(payload, file)
+        sources_api = sources_api_client(payload["rh_account"])
+        data = JSON.parse(file)
+
+        # Find Source Type
+        source_type = find_source_type(data["source_type"], sources_api)
+        # Create source with first payload
+        source = find_or_create_source(sources_api, source_type.id, data['name'], data["source"])
+
+        if source.present?
+          send_to_ingress_api(data, file)
+        else
+          raise "Failed to create Source"
+        end
+      end
+
+      def openurl(url)
```

Comment:
@slemrmartin does this just copy&paste what we're extracting to a common method over in https://github.com/ManageIQ/topological_inventory-sync/pull/8 ?

Reply:
I borrowed this from your PR :) because both PRs aren't merged, we can join it later so I used the same code to make it easier then